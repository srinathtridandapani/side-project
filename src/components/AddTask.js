import React from 'react';
import SkyLight from 'react-skylight';
import Button from '@material-ui/core/Button';
import TextField from '@material-ui/core/TextField';

class AddTask extends React.Component {
    constructor(props) {
        super(props);
        this.state = {status: '', description: ''};
    }

    handleChange = (event) => {
        this.setState(
            {[event.target.name]: event.target.value}
        );
    }

    // Save Task and close modal form
    handleSubmit = (event) => {
        event.preventDefault();
        var newTask = {status: this.state.status, description: this.state.description};
        this.props.addTask(newTask);
        this.refs.addDialog.hide();
    }

// Cancel and close modal form
    cancelSubmit = (event) => {
        event.preventDefault();
        this.refs.addDialog.hide();
    }

    render() {
        return (
            <div>
                <SkyLight hideOnOverlayClicked ref="addDialog">
                    <h3>New Task</h3>
                    <form>

                        <TextField label="Status" placeholder="Status"
                                   name="status" onChange={this.handleChange}/><br/>
                        <TextField label="Description" placeholder="Description"
                                   name="description" onChange={this.handleChange}/><br/>

                        <Button variant="outlined" color="primary"
                                onClick={this.handleSubmit}>Save</Button>
                        <Button variant="outlined" color="secondary"
                                onClick={this.cancelSubmit}>Cancel</Button>
                    </form>
                </SkyLight>
                <div>
                    <Button variant="raised" color="primary"
                            style={{'margin': '10px'}}
                            onClick={() => this.refs.addDialog.show()}>
                        New Task</Button>
                </div>
            </div>
        );
    }
}

export default AddTask;