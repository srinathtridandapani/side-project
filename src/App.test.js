import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer';
import AddTask from './components/AddTask';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders a snapshot', () => {
  const tree = renderer.create(<AddTask/>).toJSON();
  expect(tree).toMatchSnapshot();
});
