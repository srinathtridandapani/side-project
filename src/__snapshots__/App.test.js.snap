// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<div>
  <section
    className="skylight-wrapper "
  >
    <div
      className="skylight-overlay"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "rgba(0,0,0,0.3)",
          "display": "none",
          "height": "100%",
          "left": "0px",
          "position": "fixed",
          "top": "0px",
          "transitionDuration": "200ms",
          "transitionProperty": "all",
          "transitionTimingFunction": "ease",
          "width": "100%",
          "zIndex": "99",
        }
      }
    />
    <div
      className="skylight-dialog"
      style={
        Object {
          "backgroundColor": "#fff",
          "borderRadius": "2px",
          "boxShadow": "0px 0px 4px rgba(0,0,0,.14),0px 4px 8px rgba(0,0,0,.28)",
          "left": "50%",
          "marginLeft": "-25%",
          "marginTop": "-200px",
          "minHeight": "400px",
          "padding": "15px",
          "position": "fixed",
          "top": "50%",
          "transform": "scale(0)",
          "transitionDuration": "200ms",
          "transitionProperty": "transform",
          "transitionTimingFunction": "ease",
          "width": "50%",
          "zIndex": "100",
        }
      }
    >
      <a
        className="skylight-close-button"
        onClick={[Function]}
        role="button"
        style={
          Object {
            "cursor": "pointer",
            "fontSize": "1.8em",
            "position": "absolute",
            "right": "10px",
            "top": "0px",
          }
        }
      >
        Ã—
      </a>
      <h3>
        New Task
      </h3>
      <form>
        <div
          className="MuiFormControl-root-233"
        >
          <label
            className="MuiFormLabel-root-248 MuiInputLabel-root-237 MuiInputLabel-formControl-242 MuiInputLabel-animated-245"
            data-shrink={false}
          >
            Status
          </label>
          <div
            className="MuiInputBase-root-268 MuiInput-root-255 MuiInput-underline-259 MuiInputBase-formControl-269 MuiInput-formControl-256"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              className="MuiInputBase-input-278 MuiInput-input-263"
              disabled={false}
              name="status"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Status"
              required={false}
              type="text"
            />
          </div>
        </div>
        <br />
        <div
          className="MuiFormControl-root-233"
        >
          <label
            className="MuiFormLabel-root-248 MuiInputLabel-root-237 MuiInputLabel-formControl-242 MuiInputLabel-animated-245"
            data-shrink={false}
          >
            Description
          </label>
          <div
            className="MuiInputBase-root-268 MuiInput-root-255 MuiInput-underline-259 MuiInputBase-formControl-269 MuiInput-formControl-256"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              className="MuiInputBase-input-278 MuiInput-input-263"
              disabled={false}
              name="description"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Description"
              required={false}
              type="text"
            />
          </div>
        </div>
        <br />
        <button
          className="MuiButtonBase-root-311 MuiButton-root-285 MuiButton-outlined-293 MuiButton-outlinedPrimary-294"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiButton-label-286"
          >
            Save
          </span>
          <span
            className="MuiTouchRipple-root-314"
          />
        </button>
        <button
          className="MuiButtonBase-root-311 MuiButton-root-285 MuiButton-outlined-293 MuiButton-outlinedSecondary-295"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiButton-label-286"
          >
            Cancel
          </span>
          <span
            className="MuiTouchRipple-root-314"
          />
        </button>
      </form>
    </div>
  </section>
  <div>
    <button
      className="MuiButtonBase-root-311 MuiButton-root-285 MuiButton-contained-296 MuiButton-containedPrimary-297 MuiButton-raised-299 MuiButton-raisedPrimary-300"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "margin": "10px",
        }
      }
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-286"
      >
        New Task
      </span>
      <span
        className="MuiTouchRipple-root-314"
      />
    </button>
  </div>
</div>
`;
